// @pengzhanbo/utils@1.1.1/dist/cjs/is.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isDef = exports.isDate = exports.isBrowser = exports.isBoolean = exports.isArray = void 0;
exports.isEmptyObject = isEmptyObject;
exports.isFunction = isFunction;
exports.isNumber = exports.isNull = void 0;
exports.isObject = isObject;
exports.isSymbol = exports.isString = exports.isRegexp = void 0;
exports.isUndefined = isUndefined;
exports.isWindow = isWindow;
var _common = require("./common.cjs");
const isDef = v => typeof v !== "undefined";
exports.isDef = isDef;
const isBoolean = v => typeof v === "boolean";
exports.isBoolean = isBoolean;
function isFunction(v) {
  return typeof v === "function";
}
const isNumber = v => typeof v === "number";
exports.isNumber = isNumber;
const isString = v => typeof v === "string";
exports.isString = isString;
const isSymbol = v => typeof v === "symbol";
exports.isSymbol = isSymbol;
function isObject(v) {
  return (0, _common.getTypeName)(v) === "object";
}
const isArray = v => Array.isArray(v);
exports.isArray = isArray;
function isUndefined(v) {
  return typeof v === "undefined";
}
const isNull = v => (0, _common.getTypeName)(v) === "null";
exports.isNull = isNull;
const isRegexp = v => (0, _common.getTypeName)(v) === "regexp";
exports.isRegexp = isRegexp;
const isDate = v => (0, _common.getTypeName)(v) === "date";
exports.isDate = isDate;
function isEmptyObject(v) {
  if (!isObject(v)) return false;
  for (const _ in v) return false;
  return true;
}
function isWindow(v) {
  return typeof v !== "undefined" && (0, _common.getTypeName)(v) === "window";
}
const isBrowser = () => typeof window !== "undefined";
exports.isBrowser = isBrowser;
